src/messaging/ExchangeContext.cpp

```diff
@@ -140,6 +142,12 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
         SetResponseExpected(false);
     }
 
+    // Standalone acks are not application-level message sends.
+    if (err == CHIP_NO_ERROR && !isStandaloneAck)
+    {
+        MessageHandled();
```

Comment:
Currently, Messaging::SendMessageFlags::kExpectResponse is only used when App get its response within a specified timeout.  A timer is started if app specify this flag. 

Most of protocols still send messages with default flag Messaging::SendMessageFlags::kNone even a response is expected in the protocol.

Does this means the app needs to specify flag Messaging::SendMessageFlags::kExpectResponse explicitly now if there is a expected response for the outgoing message?

Reply:
> A timer is started if app specify this flag.

If `mResponseTimeout > 0`.  Which is only if `SetResponseTimeout` was called.  Though we should initialize `mResponseTimeout` to 0....  I will add that to this PR.

> Most of protocols still send messages with default flag Messaging::SendMessageFlags::kNone even a response is expected in the protocol.

Which ones?  IM, echo, CASE setup, PASE setup, message counter sync all set kExpectResponse.

> Does this means the app needs to specify flag Messaging::SendMessageFlags::kExpectResponse explicitly now if there is a expected response for the outgoing message?

Yes.